docs/manual/dynamicinterpreterload.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: page
+title: "Dynamic Interpreter Loading"
+description: ""
+group: manual
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+{% include JB/setup %}
+
+## Dynamic Interpreter Loading using REST API
+
+Zeppelin provides pluggable interpreter architecture which results in a wide and variety of the supported backend system. In this section, we will introduce **Dynamic interpreter loading** using **REST API**. Actually this concept is come up with [Zeppelin Helium Proposal](https://cwiki.apache.org/confluence/display/ZEPPELIN/Helium+proposal).
+Before the start, if you are not clear about the concept of **Zeppelin interpreter**, you can ckeck out [Overview of Zeppelin interpreter](../manual/interpreters.html) first.
+
+<br/>
+## Overview 
+In the past, Zeppelin loads interpreter binaries from `/interpreter/[interpreter_name]` directory. And they are configured by `zeppelin.interpreters` property in `conf/zeppelin-site.xml` or `ZEPPELIN_INTERPRETERS` env variables in `conf/zeppelin-env.sh`. They are loaded when Zeppelin server is starting up and stay alive until the server is down.
+However, for using 3rd party interpreter much easier, we change this way to **dynamically** loading interpreters from **Maven Repository** using **REST API**. Hopefully, below picture helps you to understand the process easily. 
+<center><img src="../assets/themes/zeppelin/img/docs-img/zeppelin_user.png" height="85%" width="85%"></center>
+
+## Load & Unload Interpreters Using REST API
+
+### 1. Load 
+You can **load** interpreters located in Maven repository using REST API, like this:
+
+( Maybe, you are unfamiliar with `[interpreter_group_name]` or `[interpreter_name]`. If so, please checkout [Interpreters in Zeppelin](../manual/interpreter.html) again. )
+
+```
+http://[zeppelin-server]:[zeppelin-port]/api/interpreter/load/[interpreter_group_name]/[interpreter_name]
+```
+The Restful method will be <code>**POST**</code>. And the parameters you need are like:
+
+  1.  **Artifact:** Maven artifact ( groupId:artifactId:version ) 
+
+  2.  **Class Name:** Package name + Interpreter class name
+
+  3. **Repository ( optional ):** Additional maven repository address
+
+For example, if you want to load `markdown` interpreter to your Zeppelin, above parameters and the URL you need may like:
```

Comment:
above parameters and the URL you need may like:
-> the parameters and URL you need may look like: